### PR TITLE
Add sal.h from Microsoft GitHub repository

### PR DESCRIPTION
**Pull request by tkreuzer**
The code is released by MS under the MIT license.
See https://github.com/microsoft/service-fabric/blob/master/src/prod/src/pal/src/internal/rt/sal.h
This allows to do static analysis with MSVC again.

### CODE REVIEW

**Comment by HBelusca** [comment]
Our sal header was broken in that regard?

File: sdk/include/psdk/sal.h [hunk 1]
```diff
@@ -1083,664 +1649,1316 @@ __ANNOTATION(SAL_nonvolatile(void));
...
+#define __deref_opt_inout                                        _SAL1_Source_(__deref_opt_inout, (), _Inout_opt_)
+#define __deref_opt_inout_ecount(size)                           _SAL1_Source_(__deref_opt_inout_ecount, (size), __deref_inout_ecount(size)                  __pre_except_maybenull)
+#define __deref_opt_inout_bcount(size)                           _SAL1_Source_(__deref_opt_inout_bcount, (size), __deref_inout_bcount(size)                  __pre_except_maybenull)
+#define __deref_opt_inout_ecount_part(size,length)               _SAL1_Source_(__deref_opt_inout_ecount_part, (size,length), __deref_inout_ecount_part(size,length)      __pre_except_maybenull)
+#define __deref_opt_inout_bcount_part(size,length)               _SAL1_Source_(__deref_opt_inout_bcount_part, (size,length), __deref_inout_bcount_part(size,length)      __pre_except_maybenull)
+#define __deref_opt_inout_ecount_full(size)                      _SAL1_Source_(__deref_opt_inout_ecount_full, (size), __deref_inout_ecount_full(size)             __pre_except_maybenull)
+#define __deref_o
```
Comment (HBelusca) [inline sdk/include/psdk/sal.h]: these lines might be removed.

Comment (gigaherz) [reply sdk/include/psdk/sal.h]: Isn't it best to leave the code as original as possible to facilitate future syncs?

Comment (tkreuzer) [reply sdk/include/psdk/sal.h]: I don't want to change anything, unless required.

Comment (HBelusca) [reply sdk/include/psdk/sal.h]: ok

**Comment by tkreuzer** [comment]
> 
> When you get this to build in gcc, ship it

Ugh... from mingw32 c++ headers:
```    const int __ret = __builtin_vsprintf(__out, __fmt, __args);```
Gonna comment out SAL1.

**Pull request merged**